src/connection_pool.rs

```diff
@@ -537,6 +535,30 @@ where
     }
 }
 
+impl<C, R, F> Deref for PoolConnection<C, R, F>
+where
+    C: Connection,
+    R: Resolver,
+    F: FnMut(&Backend) -> C + Send
+{
+    type Target = C;
+
+    fn deref(&self) -> &C {
+        &(self.connection_pair.0).1.as_ref().unwrap()
```

Comment:
Are the tuple indexes (0, 1) guaranteed to be present?

Reply:
They are. The compiler will detect and give an error if you try to use an index not present for the type. For example if I changes the first index to a 3 the compiler tells me this:

```
error[E0609]: no field `3` on type `connection_pool::types::ConnectionKeyPair<C>`
   --> src/connection_pool.rs:547:32
    |
547 |         &(self.connection_pair.3).1.as_ref().unwrap()
    |                                ^ help: a field with a similar name exists: `0`

error: aborting due to previous error
```